docs/candid/01-style-guide.md

```diff
@@ -2,3 +2,306 @@
+};
+```
+
+:::
+
+### Variants
+
+Use `lower_snake_case` for variant definitions.
+
+This has no additional benefit aside from consistency.
+
+:::danger[**Don't**]
+
+**Don't** use inconsisent or alternative casings for variant definitions.
+
+```protobuf
+type UserRole = variant {
+  Admin;
+  MODERATOR;
+  user;
+};
+```
+
+:::
+
+:::tip[**Do**]
+
+**Do** use `lower_snake_case` casing for variant definitions consisently.
+
+```protobuf
+type UserRole = variant {
+  admin;
+  moderator;
+  user;
+};
+```
+
+:::
+
+### Service methods
+
+Use `lower_snake_case` for service method names.
+
+This has no additional benefit aside from consistency.
+
+:::danger[**Don't**]
+
+**Don't** use inconsisent or alternative casings for service method definitions.
+
+```protobuf
+service : {
+  CREATE_USER : (UserRequest) -> (UserResponse);
+  UpdateUser : (UserRequest) -> (UserResponse);
+};
+```
+
+:::
+
+:::tip[**Do**]
+
+**Do** use `lower_snake_case` casing for service method definitions consisently.
+
+```protobuf
+service : {
+  create_user : (UserRequest) -> (UserResponse);
+  update_user : (UserRequest) -> (UserResponse);
+};
+```
+
+:::
+
+### Tuple field types
+
+Use `type` aliases for tuple field types.
+
+This can help greatly with readability, by allowing the alias to act as a label for the field.
+
+:::danger[**Don't**]
+
+**Don't** use built-in Candid types for tuples.
+
+```protobuf
+type UserPost = record {
+  nat64;
+  nat64;
+};
+```
+
+:::
+
+:::tip[**Do**]
+
+**Do** use type alises in tuples.
+
+```protobuf
+type UserId = nat64;
+type PostId = nat64;
+
+type UserPost = record {
+  UserId;
+  PostId;
+};
+```
+
+:::
+
+### Type aliases
+
+Use `type` aliases to define common types that are used in multiple places.
+
+This can help with readability and also make it easier to change the underlying type in the future.
+
+:::danger[**Don't**]
+
+**Don't** define common types using using built-in Candid types.
+
+```protobuf
+type User = record {
+  id : nat64;
+  name : text;
+};
+
+type Post = record {
+  id : nat64;
+  user_id : nat64;
+  content : text;
+};
+```
+
+:::
+
+:::tip[**Do**]
+
+**Do** define common types with type alises.
+
+```protobuf
+type UserId = nat64;
+type PostId = nat64;
+
+type User = record {
+  id : UserId;
+  name : text;
+};
+
+type Post = record {
+  id : PostId;
+  user_id : UserId;
+  content : text;
+};
+```
+
+:::
+
+### Request and response types
+
+Use separate request and response types for each service method. Name them `{MethodName}Request` and `{MethodName}Reponse` respectively.
+
+This allows for changes to a single service method without affecting other methods. It may be tempting to reuse common request or response types for multiple methods, but over time this can lead to a lot of messy coupling between methods. Unique request and response types also allows for readers to quickly understand the purpose of each type, without needing to explicitly check the service method that it is used for.
```

Comment:
```suggestion
This allows for changes to a single service method without affecting other methods. It may be tempting to reuse common request or response types for multiple methods, but over time this can lead to a lot of messy coupling between methods. Unique request and response types also allow readers to quickly understand the purpose of each type, without needing to explicitly check the service method that it is used for.
```